docs/parser_plugins.md

```diff
@@ -1 +1,423 @@
+        child_archives: dict[str, EntryArchive] = None,
+    ) -> None:
+        logger.info('VasprunXMLParser.parse', parameter=configuration.parameter)
+        data_parser = MetainfoParser(annotation_key='xml', data_object=Simulation())
+        XMLParser(filepath=mainfile).convert(data_parser)
+        archive.data = data_parser.data_object
+```
+
+<!-- TODO Is normalization automatically triggered here? -->
+
+This is all of the code!
+Leveraging `MappingAnnotationModel` clearly makes parsers more concise.
+Breaking down the parser class, `VasprunXMLParser`, it inherits most functionality from `MatchingParser`, including `parse`.
+The `parse` interface is rigid -represents a contract between the NOMAD base and the parser- and is worth studying a bit:
+
+The `MetainfoParser` is linked to the annotated schema. A single schema may contain multiple annotations corresponding to various file formats, e.g. `xml`, `hdf5`, `txt` for VASP. Since we are dealing with computational data, we provide the `Simulation` object as the root of our target.
+
+## From Hierarchies to Schema
+
+We conceptualize format conversion as restructuring a _hierarchical data tree_, formally known as a _directed acyclic graph_.
+The restructuring may then be defined in terms of lining up source with target nodes.
+The only missing component then are the mappings themselves.
+
+Since we have full control over our own schema, we will annotate the mappings there.
+The only relevant nodes are `ArchiveSection` attributes like `Quantity` (i.e. leaf nodes) or `SubSection` (i.e. branching nodes).
+Adding annotations then is as simple as extending a dictionary: `<section_name>.<attribute_name>.m_annotations[<tag_name>] = MappingAnnotationModel(...)`.
+
+The overall strategy is thus to annotate `path` mappings starting from `Simulation` and run over each `SubSection` up until reaching the corresponding `Quantity`.
+The most important part is for the target/NOMAD path to be fully traceable:
+any disconnections in a `path` will cut it out of the `archive`.
+
+For VASP, the `src/nomad_parser_vasp/schema_packages/` file will thus look something like the following.
+Note how you can trace continuous a path down to `simulation.program.name`.
+The mixing of multiple extension annotations (`xml` and `hdf5`), also shows at a glance how much both formats compare.
+
+<!-- TODO: extend example + add hdf5? -->
+
+```python
+from nomad_simulations.schema_packages.general import Simulation, Program
+
+Simulation.m_def.m_annotations['xml'] = MappingAnnotationModel(path='modeling')
+
+Simulation.program.m_annotations['xml'] = MappingAnnotationModel(path='.generator')
+
+Program.name.m_annotations['xml'] = MappingAnnotationModel(path='.i[?"@name"="program"]')
+```
+
+Many more approaches are at play in this example.
+Check out the "Extra" sections below for a more in-depth explanation and edge cases.
+
+### Extra: exploring Mapping Annotations
```

Comment:
I think we may want to leave this closer look out for now until we get further approval from Markus, I feel like it is too risky